test/framework/easyconfig.py

```diff
@@ -1876,6 +1892,244 @@ def test_dump_comments(self):
+        write_file(testec, '')
+        rawtxt = '\n'.join([
+            "# this is a header",
+            "#",
+            "# which may include empty comment lines",
+            "    # weirdly indented lines",
+            '  ',  # whitespace-only line, should get stripped (but no # added)
+            "# or flat out empty lines",
+            '',
+            "easyblock = 'ConfigureMake'",
+            '',
+            "name = 'thisisjustatest'  # just a test",
+            "# the version doesn't matter much here",
+            "version = '1.2.3'",
+            '',
+            "homepage = 'https://example.com'  # may not be actual homepage",
+            "description = \"\"\"Sneaky description with hashes, line #1",
+            " line #2",
+            " line without hashes",
+            "#4 (yeah, sneaky, isn't it),",
+            "end line",
+            "\"\"\"",
+            '',
+            "toolchain = SYSTEM",
+            "# after toolchain, before sources comment",
+            '',
+            "# this comment contains another #, uh-oh...",
+            "sources = ['test-1.2.3.tar.gz']",
+            "# how about # a comment with # multple additional hashes",
+            "source_urls = [",
+            "    # first possible source URL",
+            "    'https://example.com',",
+            "# annoying non-indented comment",
+            "    'https://anotherexample.com',  # fallback URL",
+            "]",
+            '',
+            "# this is a multiline comment above dependencies",
+            "# I said multiline",
+            "# multi > 3",
+            "dependencies = [",
+            "    # this dependency",
+            "# has multiple lines above it",
+            "    # some of which without proper indentation...",
+            "    ('foo', '1.2.3'),  # and an inline comment too",
+            "    ('nocomment', '4.5'),",
+            "    # last dependency, I promise",
+            "    ('last', '1.2.3'),",
+            "    # trailing comments in dependencies",
+            "    # a bit weird, but it happens",
+            "]  # inline comment after closing dependencies",
+            '',
+            "# how about comments above and in a dict value?",
+            "sanity_check_paths = {",
+            "    # a bunch of files",
+            "    'files': ['bin/foo', 'lib/libfoo.a'],",
+            "    # no dirs!",
+            "    'dirs': [],",
+            "}",
+            '',
+            "moduleclass = 'tools'",
+            "#",
+            "# trailing comment",
+            '',
+            "# with an empty line in between",
+            "# DONE!",
+        ])
+        ec = EasyConfig(None, rawtxt=rawtxt)
+
+        # check internal structure to keep track of comments
+        self.assertEqual(ec.parser._formatter.comments['above'], {
+           "    'https://anotherexample.com',": ['# annoying non-indented comment'],
```

Comment:
Shouldn't this one be in iterabove?

Reply:
Yeah, good point, it should be. Fixed in da3532be8